src/main/kotlin/it/unibo/protelis/web/execution/simulated/AlchemistVerticle.kt

```diff
@@ -0,0 +1,76 @@
+package it.unibo.protelis.web.execution.simulated
+
+import io.vertx.core.Context
+import io.vertx.core.Vertx
+import io.vertx.core.eventbus.EventBus
+import io.vertx.core.logging.Logger
+import io.vertx.core.logging.LoggerFactory
+import io.vertx.kotlin.coroutines.CoroutineVerticle
+import it.unibo.protelis.web.execution.CoroutineProtelisEngine
+import it.unibo.protelis.web.execution.EventBusProtelisObserver
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+class AlchemistVerticle : CoroutineVerticle() {
+  private val logger: Logger = LoggerFactory.getLogger(this::class.java)
+  private val engines: MutableMap<String, CoroutineProtelisEngine> = mutableMapOf()
+  private lateinit var eb: EventBus
+
+  override fun init(vertx: Vertx, context: Context) {
+    super.init(vertx, context)
+    eb = vertx.eventBus()
+  }
+
+  override suspend fun start() {
+    eb.consumer<String>(setupAddress()) { msg ->
+      val id = addressIdGen()
+      val source = msg.body()
+      val sim = SimulatedProtelisEngine()
+      engines += id to sim
+      msg.reply(id)
+      eb.consumer<Unit>(stopAddress(id)) { engines[id]?.stop() }
+      sim
+        .setup(source, EventBusProtelisObserver(eb, id))
+        .thenCompose { sim.start() }
+        .thenRun { logger.info("Simulation $id started") }
+    }
+  }
+
+  companion object {
+    private const val BASE_ADDRESS = "protelis.web.exec"
+    private const val ID_REGEX = """\S+"""
+
+    private const val INITIALIZED_SUFFIX = "init"
+    fun initializedAddress(addressId: String): String = "$BASE_ADDRESS.$addressId.$INITIALIZED_SUFFIX"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [UndocumentedPublicFunction - [initializedAddress] at /src/src/main/kotlin/it/unibo/protelis/web/execution/simulated/AlchemistVerticle.kt:44:5](https://app.codacy.com/manual/NiccoMlt/Protelis-Web/pullRequest?prid=4969516)